sdk/cosmos/azure-cosmos-encryption/src/main/java/com/azure/cosmos/encryption/EncryptionCosmosAsyncContainer.java

```diff
@@ -261,48 +375,85 @@
-            Preconditions.checkArgument(partitionKey != null, "partitionKey cannot be null for operations using "
-                + "EncryptionContainer.");
+        if (encryptionItemRequestOptions == null || encryptionItemRequestOptions.getEncryptionOptions() == null) {
+            return container.replaceItem(item, itemId, partitionKey, requestOptions);
+        }
+
+        Preconditions.checkArgument(partitionKey != null, "partitionKey cannot be null for operations using "
+            + "EncryptionContainer.");
 
-            return Mono.defer(() -> {
-                byte[] payload = cosmosSerializerToStream(item);
-                Mono<CosmosItemResponse<byte[]>> result = this.replaceItemStream(payload,
-                    itemId,
-                    partitionKey,
-                    encryptionItemRequestOptions);
 
-                return result.map(rsp -> (CosmosItemResponse<T>) this.responseFactory.createItemResponse(rsp,
-                    item.getClass()));
-            }).subscribeOn(encryptionScheduler);
+        EncryptableItem encryptableItem = Utils.as(item, EncryptableItem.class);
+        if (encryptableItem != null) {
+
+            // using (Stream streamPayload = encryptableItem.ToStream(this.CosmosSerializer))
+            byte[] streamPayload = encryptableItem.toStream(this.getItemDeserializer());
+
+            Mono<CosmosItemResponse<byte[]>> rspMono = this.replaceItemHelper(streamPayload,
+                itemId,
+                partitionKey,
+                requestOptions,
+                false);
+
+
+            return rspMono.map(
+                rsp -> {
+                    encryptableItem.setDecryptableItem(
+                        //EncryptionProcessor.BaseSerializer.FromStream<JObject>(responseMessage.Content),
+                        getItemDeserializer().parseFrom(ObjectNode.class,
+                            EncryptionModelBridgeInternal.getByteArrayContent(rsp)),
+                        this.encryptor,
+                        this.getItemDeserializer()
+                    );
+
+                    return createEncryptionItemResponse(rsp, (T) encryptableItem);
+                }
+            ).subscribeOn(encryptionScheduler);
 
         } else {
-            return container.replaceItem(item, itemId, partitionKey, requestOptions);
+            byte[] streamPayload = cosmosSerializerToStream(item);
+            Mono<CosmosItemResponse<byte[]>> response = replaceItemHelper(
+                streamPayload,
+                itemId,
+                partitionKey,
+                requestOptions,
+                true
+            );
+
+            return response
+                .publishOn(encryptionScheduler)
+                .map(
+                    rsp -> {
+                        return this.responseFactory.createItemResponse(rsp, (Class<T>) item.getClass());
+                    }
+                );
+
+            // TODO: no support for streaming
```

Comment:
Same as above - at least to me it isn't clear what this comment means. We don't support Stream Apis at all in Java,  correct?

Reply:
you are right the comment is confusing.
we don't support streaming api in java SDK.

will cleanup